test/from-mbtiles.js

```diff
@@ -0,0 +1,58 @@
+import { MBTiles } from 'mbtiles-reader'
+import { temporaryFile } from 'tempy'
+
+import assert from 'node:assert/strict'
+import fs from 'node:fs'
+import { buffer } from 'node:stream/consumers'
+import { pipeline } from 'node:stream/promises'
+import test from 'node:test'
+import { fileURLToPath } from 'node:url'
+
+import convert from '../lib/from-mbtiles.js'
+import SMPReader from '../lib/reader.js'
+
+test('convert from MBTiles', async (t) => {
+  const fixture = fileURLToPath(
+    new URL('./fixtures/plain_1.mbtiles', import.meta.url),
+  )
+  const output1 = temporaryFile()
+  const output2 = temporaryFile()
+
+  t.after(() => {
+    fs.unlinkSync(output1)
+    fs.unlinkSync(output2)
+  })
+
+  await Promise.all([
+    convert(fixture, output1),
+    pipeline(convert(fixture), fs.createWriteStream(output2)),
+  ])
+
+  for (const output of [output1, output2]) {
+    const smp = new SMPReader(output)
+    const mbtiles = new MBTiles(fixture)
+    const style = await smp.getStyle('')
+    const sourceMetadata = Object.values(style.sources)[0]
+    assert.equal(sourceMetadata.type, 'raster')
+    for (const { x, y, z, data } of mbtiles) {
```

Comment:
Maybe worth verifying that `mbtiles` is nonempty?

Reply:
done in 6b75988029bc0e03bc8959ac7168ac9a4402537a